sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/ConversationAnalysisClientLiveTests.cs

```diff
@@ -9,5 +16,93 @@ public ConversationAnalysisClientLiveTests(bool isAsync, ConversationAnalysisCli
+        private static string SpanishText = "Tendremos 2 platos de nigiri de salmón braseado.";
+        private static List<string> ExpectedOutput = new List<string>()
+        {
+            "2 plates",
+            "seared salmon nigiri",
+            "2 plates of seared salmon nigiri"
+        };
+
+        [RecordedTest]
+        public async Task AnalyzeConversation()
+        {
+            ConversationAnalysisClient client = Client;
+
+            AnalyzeConversationOptions options = new AnalyzeConversationOptions(
+               TestEnvironment.ProjectName,
+               TestEnvironment.DeploymentName,
+               EnglishText);
+
+            Response<AnalyzeConversationResult> response = await client.AnalyzeConversationAsync(options);
+
+            Assert.That(response.GetRawResponse().Status, Is.EqualTo(200));
+            Assert.That(response.Value.Prediction.TopIntent, Is.EqualTo("Order"));
+            Assert.That(response.Value.Prediction.ProjectKind, Is.EqualTo(ProjectKind.Conversation));
+        }
+
+        [RecordedTest]
+        public async Task AnalyzeConversationWithLanguage()
+        {
+            ConversationAnalysisClient client = Client;
+
+            AnalyzeConversationOptions options = new AnalyzeConversationOptions(
+               TestEnvironment.ProjectName,
+               TestEnvironment.DeploymentName,
+               SpanishText)
+            {
+                Language = "es"
+            };
+
+            Response<AnalyzeConversationResult> response = await client.AnalyzeConversationAsync(options);
+
+            Assert.That(response.GetRawResponse().Status, Is.EqualTo(200));
+            Assert.That(response.Value.Prediction.TopIntent, Is.EqualTo("Order"));
+            Assert.That(response.Value.Prediction.ProjectKind, Is.EqualTo(ProjectKind.Conversation));
+        }
+
+        [RecordedTest]
+        public async Task AnalyzeConversationsDeepstack()
+        {
+            ConversationAnalysisClient client = Client;
+
+            AnalyzeConversationOptions options = new AnalyzeConversationOptions(
+               TestEnvironment.ProjectName,
+               TestEnvironment.DeploymentName,
+               EnglishText);
+
+            Response<AnalyzeConversationResult> response = await client.AnalyzeConversationAsync(options);
+
+            DeepstackPrediction deepstackPrediction = response.Value.Prediction as DeepstackPrediction;
+
+            Assert.That(response.GetRawResponse().Status, Is.EqualTo(200));
+            Assert.That(response.Value.Prediction.ProjectKind, Is.EqualTo(ProjectKind.Conversation));
+
+            Assert.That(deepstackPrediction.TopIntent, Is.EqualTo("Order"));
+
+            IList<string> entitiesText = deepstackPrediction.Entities.Select(entity => entity.Text).ToList();
+            Assert.That(entitiesText, Has.Count.AtLeast(1));
+            Assert.That(entitiesText, Is.SubsetOf(ExpectedOutput));
```

Comment:
There is an option to assert on only the answer which is: 
```C#
private static List<string> ExpectedOutput = new List<string>()
{
       "2 plates",
       "seared salmon nigiri",
};
```

```C#
Assert.That(entitiesText, Has.Count.EqualTo(2));
Assert.That(entitiesText, Is.EqualTo(ExpectedOutput));
```

But I wasn't sure which is better.

Reply:
If `Is.EqualTo` is doing a deep assertion, that would be better. That said, if the order isn't always the same, you could also do a pair of "Contains" checks (I forget on which helper that's located) for each expected item so it ignores the order. We do that in a lot of places where the order isn't always deterministic (if it doesn't need to be).

Reply:
I tried changing the order in `ExpectedOne` and `Is.SubsetOf` worked perfectly. I think it's good to go, what do you think?

Reply:
Hmm... I think I'd prefer an exact (order not important) match. If there's more of one or another, that could mean a problem. Either `CollectionAssert.AreEquivalent` or some form of `IsEquivalent` when using `Assert.That` would be better, it seems.

Reply:
I misunderstood that, I get your point now. Will change it right away.